src/Zed/Persistence/Repository/RepositoryPropelQueryJoinRule.php

```diff
@@ -0,0 +1,198 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace ArchitectureSniffer\Zed\Persistence\Repository;
+
+use ArchitectureSniffer\Module\Transfer\ModuleTransfer;
+use ArchitectureSniffer\PropelQuery\PropelQueryFacade;
+use PHPMD\AbstractNode;
+use PHPMD\Rule\MethodAware;
+
+class RepositoryPropelQueryJoinRule extends AbstractRepositoryRule implements MethodAware
+{
+    public const RULE = 'All joins must be declared in DocBlock';
+
+    /**
+     * @var \ArchitectureSniffer\PropelQuery\PropelQueryFacade
+     */
+    protected $facade;
+
+    /**
+     * @param \PHPMD\AbstractNode $node
+     *
+     * @return void
+     */
+    public function apply(AbstractNode $node): void
+    {
+        if (!$this->isRepository($node)) {
+            return;
+        }
+
+        $joinNames = $this->getFacade()->getJoinNames($node);
+
+        if ($joinNames === []) {
+            return;
+        }
+
+        foreach ($joinNames as $joinName) {
+            $this->applyRule($joinName, $node);
+        }
+    }
+
+    /**
+     * @param string $joinName
+     * @param \PHPMD\AbstractNode $node
+     *
+     * @return void
+     */
+    protected function applyRule(string $joinName, AbstractNode $node): void
+    {
+        $rootPath = $this->getFacade()->getRootApplicationFolderPathByFilePath($node->getFileName());
+
+        $docBlockModuleNames = $this->getFacade()->getDocBlockModules($node);
+
+        $moduleNames = $this->getModuleNames($joinName, $node, $docBlockModuleNames);
+
+        $moduleTransfers = $this->getFacade()->findModulesByNames($moduleNames, $rootPath);
+
+        $tableTransfers = [];
+
+        foreach ($moduleTransfers as $moduleTransfer) {
+            $tableTransfers = $this->getTableTransfers($moduleTransfer, $tableTransfers);
+        }
+
+        $moduleName = $this->findJoinModuleNameFromTables($joinName, $tableTransfers);
```

Comment:
`getDependentModuleNameFromTables ` as it will not return null and there has to be a dependent module